mlir/include/mlir/Dialect/WebAssemblySSA/IR/WebAssemblySSAOps.td

```diff
@@ -0,0 +1,679 @@
+  let summary = "Set local to given value";
+  let arguments = (ins WasmSSA_LocalRef: $localVar,
+                       WasmSSA_ValType: $value);
+  let hasVerifier = 1;
+  let assemblyFormat = "$localVar `:` type($localVar) `to` $value `:` type($value) attr-dict";
+}
+
+def WasmSSA_LocalTeeOp : WasmSSA_Op<"local_tee", [
+    DeclareOpInterfaceMethods<InferTypeOpInterface>]> {
+  let summary = "Set local to value and return the operand.";
+  let arguments = (ins WasmSSA_LocalRef: $localVar,
+                       WasmSSA_ValType: $value);
+  let results = (outs WasmSSA_ValType: $result);
+  let hasVerifier = 1;
+  let assemblyFormat = "$localVar `:` type($localVar) `to` $value `:` type($value) attr-dict";
+}
+
+def WasmSSA_MemOp : WasmSSA_Op<"memory", [Symbol]> {
+  let summary= "WebAssembly memory definition";
+  let arguments = (ins SymbolNameAttr: $sym_name,
+                     WasmSSA_LimitTypeAttr: $limits,
+                     OptionalAttr<StrAttr>:$sym_visibility);
+  let builders = [
+    OpBuilder<(ins
+    "llvm::StringRef":$symbol,
+    "LimitType":$limit)>
+  ];
+}
+
+def WasmSSA_MemImportOp : WasmSSA_Op<"import_mem", [Symbol, WasmSSAImportOpInterface]> {
+  let summary = "Importing a memory";
+  let arguments = (ins SymbolNameAttr: $sym_name,
+                     StrAttr: $moduleName,
+                     StrAttr: $importName,
+                     WasmSSA_LimitTypeAttr: $limits,
+                     OptionalAttr<StrAttr>:$sym_visibility);
+  let extraClassDeclaration = [{
+     bool isDeclaration() const { return true; }
+   }];
+  let builders = [OpBuilder<(ins
+    "llvm::StringRef":$symbol,
+    "llvm::StringRef":$moduleName,
+    "llvm::StringRef":$importName,
+    "wasmssa::LimitType":$limits)>];
+  let assemblyFormat = "$importName `from` $moduleName `as` $sym_name attr-dict";
+}
+
+def WasmSSA_TableOp : WasmSSA_Op<"table", [Symbol]> {
+  let summary= "WebAssembly table value";
+  let arguments = (ins SymbolNameAttr: $sym_name,
+                     WasmSSA_TableTypeAttr: $type,
+                     OptionalAttr<StrAttr>:$sym_visibility);
+  let builders = [OpBuilder<(ins
+      "llvm::StringRef":$symbol,
+      "wasmssa::TableType":$type)>];
+}
+
+def WasmSSA_TableImportOp : WasmSSA_Op<"import_table", [Symbol, WasmSSAImportOpInterface]> {
+  let summary = "Importing a table";
+  let arguments = (ins SymbolNameAttr: $sym_name,
+                     StrAttr: $moduleName,
+                     StrAttr: $importName,
+                     WasmSSA_TableTypeAttr: $type,
+                     OptionalAttr<StrAttr>:$sym_visibility);
+  let extraClassDeclaration = [{
+    bool isDeclaration() const { return true; }
+  }];
+  let assemblyFormat = "$importName `from` $moduleName `as` $sym_name attr-dict";
+  let builders = [OpBuilder<(ins
+      "llvm::StringRef":$symbol,
+      "llvm::StringRef":$moduleName,
+      "llvm::StringRef":$importName,
+      "wasmssa::TableType":$type)>];
+}
```

Comment:
Are such operation directly 1:1 to the WASM specification? If not I would think we should provide a complete `let description =` field to careful describe their semantics, and provide syntax example.

Reply:
Most of them are but I'll add descriptions for the ones that aren´t